Error on preparser loading

**Describe the bug**
presentation did not run if we place a `preparser.ts` 

**To Reproduce**
Steps to reproduce the behavior:
1. npm init slidev
2. add `preparser.ts` in setup directory with following minimal content:
```
import { definePreparserSetup } from '@slidev/types'

export default definePreparserSetup(() => {
  return [
    {
      transformRawLines(lines) {

      }
    },
  ]
})
```
3. start slidev
`npm run dev `
4. See the error
``` 
node:internal/modules/cjs/loader:415
      const err = new Error(
                  ^

Error: Cannot find module '/home/alex/test_pres/test/slidev/node_modules/@slidev/types/dist/index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (node:internal/modules/cjs/loader:415:19)
    at Module._findPath (node:internal/modules/cjs/loader:665:18)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1034:27)
    at Function.resolve (node:internal/modules/helpers:136:19)
    at _resolve (/home/alex/test_pres/test/slidev/node_modules/jiti/dist/jiti.js:1:250334)
    at jiti (/home/alex/test_pres/test/slidev/node_modules/jiti/dist/jiti.js:1:252917)
    at /home/alex/test_pres/test/slidev/setup/preparser.ts:1:176
    at evalModule (/home/alex/test_pres/test/slidev/node_modules/jiti/dist/jiti.js:1:255614)
    at jiti (/home/alex/test_pres/test/slidev/node_modules/jiti/dist/jiti.js:1:253542)
    at loadSetups (file:///home/alex/test_pres/test/slidev/node_modules/@slidev/cli/dist/chunk-CYLMMBRG.mjs:24:70) {
  code: 'MODULE_NOT_FOUND',
  path: '/home/alex/test_pres/test/slidev/node_modules/@slidev/types/package.json',
  requestPath: '@slidev/types'
}
```

**Desktop (please complete the following information):**
 - OS: arch linux
 - Browser: chromium 117.0.5938.88
 - Slidev version:  v0.43.12

--- a/packages/types/package.json
+++ b/packages/types/package.json
@@ -11,7 +11,7 @@
     "url": "https://github.com/slidevjs/slidev"
   },
   "bugs": "https://github.com/slidevjs/slidev/issues",
-  "main": "dist/index.js",
+  "main": "dist/index.mjs",
   "module": "dist/index.mjs",
   "types": "index.d.ts",
   "files": [

